src/modules/profiles/application/services/organization.service.ts

```diff
@@ -14,32 +17,31 @@ export class OrganizationService implements ProfileService {
     private readonly organizationRepository: IOrganizationRepository
   ) {}
 
-  private convertOperatingHours(dto: OrganizationProfileDataDto): OperatingHours[] {
-    const daysMap = {
-      sunday: { dayNumber: 1, dayOfWeek: 'ראשון' },
-      monday: { dayNumber: 2, dayOfWeek: 'שני' },
-      tuesday: { dayNumber: 3, dayOfWeek: 'שלישי' },
-      wednesday: { dayNumber: 4, dayOfWeek: 'רביעי' },
-      thursday: { dayNumber: 5, dayOfWeek: 'חמישי' },
-      friday: { dayNumber: 6, dayOfWeek: 'שישי' },
-      saturday: { dayNumber: 7, dayOfWeek: 'שבת' }
-    };
-
-    const operatingHours: OperatingHours[] = [];
-    
-    if (dto.operatingHours) {
-      for (const [day, hours] of Object.entries(dto.operatingHours)) {
-        if (hours && daysMap[day]) {
-          const timeRanges = hours.ranges.map(range => new TimeRange(range));
-          operatingHours.push(new OperatingHours({
-            ...daysMap[day],
-            timeRanges
-          }));
-        }
-      }
-    }
-
-    return operatingHours;
+  private convertWeeklySchedule(dto: OrganizationProfileDto): WeeklySchedule | undefined {
```

Comment:
We were missing casting - now it should work :)
```suggestion
	private convertWeeklySchedule(dto: OrganizationProfileDto): WeeklySchedule | undefined {
	  if (!dto.weeklySchedule) return undefined;
	
	  const daysOfWeek: (keyof WeeklySchedule)[] = [
	    'sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday'
	  ];
	
	  const schedule = daysOfWeek.reduce((acc, day) => {
	    if (dto.weeklySchedule[day]) {
	      (acc[day] as DaySchedule) = this.convertDaySchedule(dto.weeklySchedule[day]);
	    }
	    return acc;
	  }, {} as Partial<WeeklySchedule>);
	
	  return new WeeklySchedule(schedule);
	}
```